### PR TITLE
Added multi cluster blog

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 90f411abd846b088e14f18a310ebb4a2c660dfc5 at: https://62b0ad86d9b48a674ad432a6--layer5.netlify.app

**Comment by Revolyssup** [comment]
@debo19  In the preview, I am not seeing the images although I see them locally. Can you figure why?

**Comment by l5io** [comment]
🚀 Preview for commit 7eb27f04fff718ed9e5d0aa5b4618a7789bafa2e at: https://62b0b20d93f7ca7073b62612--layer5.netlify.app

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+published: true
+---
+
+import { BlogWrapper } from "../../Blog.style.js";
+
+<BlogWrapper>
+
+# Managing multiple clusters with Meshery
+<p align="center">
+  <img src="./thumbnail.png" />
```
Comment (debo19) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:23]: Can you try importing all images on top and then referencing them here?

**Comment by leecalcote** [comment]
@Revolyssup, I'll assist with a new blog thumbnail image. Please include `alt` attributes for each `` tag.

**Comment by l5io** [comment]
🚀 Preview for commit 7ebcf235fddf1c544bfe2c38367636d6d412362d at: https://62b0bed0b9dd507437334e7e--layer5.netlify.app

**Comment by Revolyssup** [comment]
@debo19  Thank you that worked!

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx (hunk 1 above)
Comment (Revolyssup) [reply src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:30]: @debo19 Thank you that worked!

**Comment by l5io** [comment]
🚀 Preview for commit 08db765d1b81f6de4535d333ccb55938b96e2788 at: https://62b0c4a0446d527da9d29bc4--layer5.netlify.app

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+tags:
+ - Meshery
+ - Multi-cluster
+ - Kubernetes
+type: Blog
+published: true
+---
+
+import { BlogWrapper } from "../../Blog.style.js";
+import C1 from "./c1.png";
```
[the remark this answers is not in the document]
Comment (Revolyssup) [reply src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:18]: okay I will update the names

**Comment by l5io** [comment]
🚀 Preview for commit 534bb2ea7580cef11d9d4da18cb6191a3ab01686 at: https://62b35d069dd8370ea3d578d0--layer5.netlify.app

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+## Using multi cluster with Meshery
+From a client's perspective, there are two uses of the multi context feature in general. While deploying a Meshmap design or performing any other operation on their cluster(s), selecting any number of kubernetes contexts will allow them to uniformly and parallely perform the operation across the clusters. And while visualizing the state of their cluster(s), the same context switcher will allow them to filter across the clusters whose view they want to see.
```
Comment (debo19) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:36]:
```suggestion
From a client's perspective, there are two uses of the multi context feature in general. While deploying a MeshMap design or performing any other operation on their cluster(s), selecting any number of kubernetes contexts will allow them to uniformly and parallely perform the operation across the clusters. And while visualizing the state of their cluster(s), the same context switcher will allow them to filter across the clusters whose view they want to see.
```

**Review by debo19 — CHANGES_REQUESTED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit f03941b76e8d377109c5f0c09049a3f39e036426 at: https://62b369136ec793189e6de5a6--layer5.netlify.app

**Comment by Revolyssup** [comment]
> @Revolyssup I've made some small updates. @leecalcote with the new thumbnail image, this should be good to go.
Ohhh it looks pretty now. Thnx @debo19

**Comment by l5io** [comment]
🚀 Preview for commit e7af03b39d0d41da73fd57dac43dc1b246206166 at: https://62bf8a8f91bd054f39572e69--layer5.netlify.app

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+## Using multi cluster with Meshery
+From a client's perspective, there are two uses of the multi context feature in general. While deploying a <Link to="/meshmap">MeshMap</Link> design or performing any other operation on their cluster(s), selecting any number of kubernetes contexts will allow them to uniformly and parallely perform the operation across the clusters. And while visualizing the state of their cluster(s), the same context switcher will allow them to filter across the clusters whose view they want to see.
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:41]: Capitalize all instances of Kubernetes. 

Expand on what the Meshery Operator is.

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+Just before applying the operation, users will be prompted with a confirmation modal which will provide the information about which cluster(s) that operation will be performed against. As the User interface improves, this same modal will also convey more useful information about the operation they are going to perform.
+
+<p align="center">
+  <img src={Deploy} alt="deploy modal"  class="image-center-shadow" />
+</p>
+
+
+<br/>
+
+### Using Meshmap visualizer
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:62]: MeshMap is a portmanteau. Both "M"s are capitalized. Capitalize "Visualizer".

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+All cluster specific operations are now applied over a number of clusters uniformly. So if you have 10 clusters to manage and 8 of those start with the exact same set of pods, deployments, service mesh, etc then <Link to="/meshery">Meshery</Link> can help you to apply these operations quickly and easily.
+
+
+It is as simple as selecting the specific cluster(s) from the kubernetes context switcher in the navbar, and then applying whatever operation you wanted to, whether that be deploying a sample app, a service mesh, or a <Link to="/meshmap">MeshMap</Link> design.
+
+<p align="center">
+  <img src={Switcher} alt="context switcher" class="image-center-shadow" />
+</p>
+
+Just before applying the operation, users will be prompted with a confirmation modal which will provide the information about which cluster(s) that operation will be performed against. As the User interface improves, this same modal will also convey more useful information about the operation they are going to perform.
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:53]: Avoid 3rd-person form and shift to 2nd person with "you", not "users".

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+</p>
+
+
+### Managing Meshery on multiple clusters
+
+Users can perform cluster related operations from the settings page like adding more clusters, removing data from existing clusters and removing existing clusters.
+
+<img src={Settings} alt="Settings page" class="slides-right" />
+
+<Link to="/meshery">Meshery</Link> also deploys its operator across the cluster it’s about to manage. This operator manages the lifecycle of a NATS server and MeshSync. MeshSync pumps the blood into Meshery’s core, in other words, it is responsible for watching all different types of resources by establishing a watch stream over each of them. MeshSync then pumps that data into the NATS server, of which Meshery server itself is a client. From there, Meshery server gets all the relevant data related to activities in the cluster.
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:79]: "it's operator"? This major architectural component has a name and page to link to.

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+By default <Link to="/meshery">Meshery</Link> wants to be greedy and as much aware about your infrastructure as possible so it deploys its operator across each detected cluster. But you can fine tune this configuration by going over each one of them from the table as shown.
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:81]: comma needed after "by default".

Rephrase "wants to be greedy". Speak to how Meshery's behavior benefits you.

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+<img src={Cluster} alt="cluster management" class="image-center-shadow" />
+
+If you disconnect your cluster and do not want to persist the data from that cluster then you can perform a fine grain deletion by deleting all MeshSync data(which are the kubernetes objects) for that specific cluster.
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:85]: Add a space between the parenthesis: "data(which"

"fine-grained"

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+ 
+<img src={Flush} alt="flushing meshsync data"  class="image-center-shadow" />
+
+
+
+
+
+## Future of multi cluster
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:93]: hyphenate

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+<Link to="/meshery">Meshery</Link> as an extension point to your infrastructure provides out of the box value by adding components which can be kubernetes specific, service mesh specific or custom components to add new functionality. We can now add multi-cluster specific components to provide more abstraction. This model can be used along with Meshery’s multi-mesh capabilities to give an overall multi-mesh multi-cluster experience to the user. For instance, your Istio service mesh spanning across multiple clusters can be abstracted and managed by Meshery using custom components such as VirtualGateway and VirtualDestinationRules. In this case, Meshery’s Istio adapter will handle the logic of converting a VirtualGateway into gateways across the clusters. This abstraction provides high value by powering the service mesh to span across the clusters while the Ops team can configure the mesh with minimal effort.
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:95]: Don't link to Meshery this many times.

Comment (leecalcote) [reply src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:95]: Internal links are great. Find other things to link to, like the mention of Istio. We have a number of articles under `/resources`.

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+
+
+Just like the above example, many such extension points are in Meshery to add logic into and add useful functionality. And as these extension points actually extend, Meshery will continue to give more and more power to your cloud native infrastructure.
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:98]: I think that I included a reference to extension points above. If not, add one. 

Either capitalize "extension points" or name them "Meshery extension points".

Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:98]: It's not "the above example", but "the example above".

Better yet, don't make these kinds of references, instead add a caption to the image and refer to it by caption name / figure number.

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 13]
```diff
@@ -0,0 +1,99 @@
+---
+title:  Multi-Cluster Kubernetes Management with Meshery
+subtitle: Wrangling your services one cluster at-a-time
+date: 2022-07-02 08:00:00 -0630
+author: Ashish Tiwari
+description: "Manage all of your Kubernetes clusters with the cloud native management plane, Meshery. Learn how Meshery makes connecting, discovering, and configuring multiple clusters a breeze."
+thumbnail: ./multi-cluster-kubernetes-management-with-meshery.png
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:7]: Delete the old graphic.

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+
+<BlogWrapper>
+
+<div class="intro">
+<p>
+From multi-mesh to now multi-cluster, <Link to="/meshery">Meshery</Link> is continuously expanding its capability to give developers, operators, and security engineers more control over their infrastructure. In this post, we'll take a look behind the scenes at how each component in Meshery's architecture plays a role in the management of many Kubernetes clusters.
+</p>
+</div>
+
+## Philosophy behind Meshery's multi-cluster management approach
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:36]: There's not much philosophy offered as a prelude, but more a straight jump to behind-the-scenes concepts that readers have no context for. Please lead the reader in first.

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+<br/>
+
+### Using Meshmap visualizer
+You can switch between views of your cluster in visualizer mode while using <Link to="/meshmap">MeshMap</Link>.
+
+<p align="center">
+  <img src={Cluster2} alt="visualizer showing data of context1" class="slides-right" />
+</p>
+<p align="center">
+  <img src={Cluster1} alt="visualizer showing data of context2" class="slides-left" />
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:69]: A poor second `alt` description (and first, come to think of it). Please be more descriptive in both `alt`s.

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+</p>
+
+
+### Managing Meshery on multiple clusters
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:73]: Do users care more about managing Meshery or more about managing Kubernetes? Consider rephrasing.

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+
+Users can perform cluster related operations from the settings page like adding more clusters, removing data from existing clusters and removing existing clusters.
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:75]: "removing data from clusters"? Why? Please use a more compelling example; e.g. "undeploy".

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx (hunk 7 above)
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:79]: What's the "NATS Server"? Use "Meshery Broker" - https://docs.meshery.io/concepts/architecture/broker

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+
+<img src={Settings} alt="Settings page" class="slides-right" />
+
+<Link to="/meshery">Meshery</Link> also deploys its operator across the cluster it’s about to manage. This operator manages the lifecycle of a NATS server and MeshSync. MeshSync pumps the blood into Meshery’s core, in other words, it is responsible for watching all different types of resources by establishing a watch stream over each of them. MeshSync then pumps that data into the NATS server, of which Meshery server itself is a client. From there, Meshery server gets all the relevant data related to activities in the cluster.
+
+By default <Link to="/meshery">Meshery</Link> wants to be greedy and as much aware about your infrastructure as possible so it deploys its operator across each detected cluster. But you can fine tune this configuration by going over each one of them from the table as shown.
+
+<img src={Cluster} alt="cluster management" class="image-center-shadow" />
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:83]: Perhaps, a better `alt` is "Kubernetes multi-cluster management with Meshery".

SEO matters and is half the reason for this post.

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+
+If you disconnect your cluster and do not want to persist the data from that cluster then you can perform a fine grain deletion by deleting all MeshSync data(which are the kubernetes objects) for that specific cluster.
+ 
+<img src={Flush} alt="flushing meshsync data"  class="image-center-shadow" />
```
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:87]: "MeshSync"

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx (hunk 11 above)
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:95]: Consider hyphenating "out-of-the-box"

File: src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx (hunk 12 above)
Comment (leecalcote) [inline src/collections/blog/2022/2022-20-06-managing-multiple-clusters-with-meshery/index.mdx:98]: Consider another phrase other than "actually extend".

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit 0b0db0931bdc32d1a508afeca08083f3042ced79 at: https://62c87ab4174d745ff6776707--layer5.netlify.app

**Comment by Nikhil-Ladha** [comment]
I doubt if the changes requested above have been addressed. Updates on it @Revolyssup ?

**Comment by Revolyssup** [comment]
> I doubt if the changes requested above have been addressed. Updates on it @Revolyssup ?

Oh I missed it. I will be making the changes in few minutes.

**Comment by Revolyssup** [comment]
@leecalcote  @Nikhil-Ladha  I have made the asked changes.

**Comment by l5io** [comment]
🚀 Preview for commit 6d1be9c1fcb78f4935578b5743f14fdfff75b8bc at: https://62dedbf5c82bc25e77dd2de0--layer5.netlify.app

**Review by leecalcote — APPROVED** [review]
REady

**Pull request merged**